var/spack/repos/builtin/packages/netcdf-fortran/package.py

```diff
@@ -116,6 +122,41 @@ def configure_args(self):
             config_args.append('FC=%s' % self.spec['mpi'].mpifc)
             config_args.append('F77=%s' % self.spec['mpi'].mpif77)
 
+        if '+dap' in self.spec:
+            config_args.append('--enable-dap')
+        else:
+            config_args.append('--disable-dap')
+
+        if '+pic' in self.spec:
+            config_args.append('--with-pic')
+        else:
+            config_args.append('--without-pic')
```

Comment:
This is a tricky one. Currently, we support:
1. Build only static version of the library without PIC: `spack install netcdf-fortran~shared~pic`.
2. Build only static version of the library with PIC: `spack install netcdf-fortran~shared+pic`.
3. Build the static version without PIC and the shared one: `spack install netcdf-fortran+shared~pic`.
4. Build the static version with PIC and the shared one: `spack install netcdf-fortran+shared+pic`.

Now, there are two ways to support this, each having its pros and cons:
1. Inject `self.compiler.pic_flag` into C and Fortran compiler flags. This is the implemented approach (see function `flag_handler`). The advantage of this approach is that it's universal, we can apply it to non-Libtool packages too. It's also easier for us to support compilers that are not supported by Libtool (the latest version of the tool is 5 years old after all): we add a new compiler in Spack, set property `pic_flag` accordingly and things work even if Libtool does not know the flag that enables PIC for this compiler. The biggest disadvantage is that we have a common `pic_flag` value for C and Fortran compilers and assume that it works for both. However, this might not be the case for mixed toolchains, e.g. NAG/GCC. This is why I always try to make sure that there is a way to avoid the injection of `pic_flag` from the command line (in this case, `spack install netcdf-fortran~pic`).
2. We can use `--with/without-pic` configure option introduced by Libtool. First, as I have already mentioned, we have to assume that Libtool knows the right flag that enables PIC for a particular compiler. Second, in general (and in the case of `netcdf-fortran`), it's impossible to build shared library when the package is configured with the option `--without-pic`: Libtool simply drops PIC flag at all, even for objects that become part of the shared library. Therefore, we should not specify `--without-pic` if we want to build the shared version of the library. The advantage of this approach is that it potentially solves the problem with mixed toolchains: Libtool treats C and Fortran compilers independently and sets PIC flags for each one properly. However, there is a pitfall: Libtool relies on the names of the compiler executables (i.e. `gcc`, `nagfor`, `ifort`) when checking which compiler it works with. In some cases, the real names are hidden by the MPI wrappers: `mpif90` and `mpicc`, which makes it impossible for Libtool (without additional patches) to figure out the right way to set PIC flags.

I think that it's better to keep the current implementation.

Reply:
Ok, that makes sense. I will remove my implementation in favor of the existing implementation.